src/App.tsx

```diff
@@ -1,27 +1,75 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  // eslint-disable-next-line
-  console.log(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
+type State = {
+  isClockVisible: boolean;
+  name: string | number;
 };
 
+class App extends React.Component<{}, State> {
+  state = {
+    isClockVisible: true,
+    name: 'React Clock',
+  };
+
+  componentDidUpdate(_prevProps: {}, prevState: State) {
+    if (prevState.name !== this.state.name) {
+      // eslint-disable-next-line no-console
+      console.log(`Clock was renamed from ${prevState.name} to ${this.state.name}`);
+    }
+  }
+
+  randomizeClockName = () => {
+    this.setState({ name: Math.floor(Math.random() * 100) });
+  };
+
+  changeClockVisibility = () => {
+    if (this.state.isClockVisible) {
+      this.setState({ isClockVisible: false });
+    } else {
+      this.setState({ isClockVisible: true });
+    }
```

Comment:
or you can set the opposite value using the functional setState and ! operator

Reply:
> please add Clock component to the PR

Oops